File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-doc-5-4/EndUserExceptionDeclaration.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class EndUserExceptionDeclaration : ISerialized
    {
        public EndUserExceptionDeclaration(XmlNode xmlNode)
        {
            XmlNodeList creditDocumentNodeList = xmlNode.SelectNodes("creditDocument");
            
            if (creditDocumentNodeList != null)
            {
                this.creditDocument_ = new List<CreditDocument>();
                foreach (XmlNode item in creditDocumentNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            creditDocumentIDRef_ = item.Attributes["id"].Value;
                            creditDocument_.Add(new CreditDocument(item));
                            IDManager.SetID(creditDocumentIDRef_, creditDocument_[creditDocument_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            creditDocumentIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        creditDocument_.Add(new CreditDocument(item));
                        }
                    }
                    else
                    {
                        creditDocument_.Add(new CreditDocument(item));
                    }
                }
            }
            
        
            XmlNodeList organizationCharacteristicNodeList = xmlNode.SelectNodes("organizationCharacteristic");
            
            if (organizationCharacteristicNodeList != null)
            {
                this.organizationCharacteristic_ = new List<OrganizationCharacteristic>();
                foreach (XmlNode item in organizationCharacteristicNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            organizationCharacteristicIDRef_ = item.Attributes["id"].Value;
                            organizationCharacteristic_.Add(new OrganizationCharacteristic(item));
                            IDManager.SetID(organizationCharacteristicIDRef_, organizationCharacteristic_[organizationCharacteristic_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            organizationCharacteristicIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        organizationCharacteristic_.Add(new OrganizationCharacteristic(item));
                        }
                    }
                    else
                    {
                        organizationCharacteristic_.Add(new OrganizationCharacteristic(item));
                    }
                }
            }
            
        
            XmlNodeList transactionCharacteristicNodeList = xmlNode.SelectNodes("transactionCharacteristic");
            
            if (transactionCharacteristicNodeList != null)
            {
                this.transactionCharacteristic_ = new List<TransactionCharacteristic>();
                foreach (XmlNode item in transactionCharacteristicNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            transactionCharacteristicIDRef_ = item.Attributes["id"].Value;
                            transactionCharacteristic_.Add(new TransactionCharacteristic(item));
                            IDManager.SetID(transactionCharacteristicIDRef_, transactionCharacteristic_[transactionCharacteristic_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            transactionCharacteristicIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        transactionCharacteristic_.Add(new TransactionCharacteristic(item));
                        }
                    }
                    else
                    {
                        transactionCharacteristic_.Add(new TransactionCharacteristic(item));
                    }
                }
            }
            
        
            XmlNodeList supervisorRegistrationNodeList = xmlNode.SelectNodes("supervisorRegistration");
            
            if (supervisorRegistrationNodeList != null)
            {
                this.supervisorRegistration_ = new List<SupervisorRegistration>();
                foreach (XmlNode item in supervisorRegistrationNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            supervisorRegistrationIDRef_ = item.Attributes["id"].Value;
                            supervisorRegistration_.Add(new SupervisorRegistration(item));
                            IDManager.SetID(supervisorRegistrationIDRef_, supervisorRegistration_[supervisorRegistration_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            supervisorRegistrationIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        supervisorRegistration_.Add(new SupervisorRegistration(item));
                        }
                    }
                    else
                    {
                        supervisorRegistration_.Add(new SupervisorRegistration(item));
                    }
                }
            }
            
        
        }
        
    
        #region CreditDocument_
        private List<CreditDocument> creditDocument_;
        public List<CreditDocument> CreditDocument_
        {
            get
            {
                if (this.creditDocument_ != null)
                {
                    return this.creditDocument_; 
                }
                else if (this.creditDocumentIDRef_ != null)
                {
                    return this.creditDocument_; 
                }
                else
                {
                      return this.creditDocument_; 
                }
            }
            set
            {
                if (this.creditDocument_ != value)
                {
                    this.creditDocument_ = value;
                }
            }
        }
        #endregion
        
        public string creditDocumentIDRef_ { get; set; }
        #region OrganizationCharacteristic_
        private List<OrganizationCharacteristic> organizationCharacteristic_;
        public List<OrganizationCharacteristic> OrganizationCharacteristic_
        {
            get
            {
                if (this.organizationCharacteristic_ != null)
                {
                    return this.organizationCharacteristic_; 
                }
                else if (this.organizationCharacteristicIDRef_ != null)
                {
                    return this.organizationCharacteristic_; 
                }
                else
                {
                      return this.organizationCharacteristic_; 
                }
            }
            set
            {
                if (this.organizationCharacteristic_ != value)
                {
                    this.organizationCharacteristic_ = value;
                }
            }
        }
        #endregion
        
        public string organizationCharacteristicIDRef_ { get; set; }
        #region TransactionCharacteristic_
        private List<TransactionCharacteristic> transactionCharacteristic_;
        public List<TransactionCharacteristic> TransactionCharacteristic_
        {
            get
            {
                if (this.transactionCharacteristic_ != null)
                {
                    return this.transactionCharacteristic_; 
                }
                else if (this.transactionCharacteristicIDRef_ != null)
                {
                    return this.transactionCharacteristic_; 
                }
                else
                {
                      return this.transactionCharacteristic_; 
                }
            }
            set
            {
                if (this.transactionCharacteristic_ != value)
                {
                    this.transactionCharacteristic_ = value;
                }
            }
        }
        #endregion
        
        public string transactionCharacteristicIDRef_ { get; set; }
        #region SupervisorRegistration_
        private List<SupervisorRegistration> supervisorRegistration_;
        public List<SupervisorRegistration> SupervisorRegistration_
        {
            get
            {
                if (this.supervisorRegistration_ != null)
                {
                    return this.supervisorRegistration_; 
                }
                else if (this.supervisorRegistrationIDRef_ != null)
                {
                    return this.supervisorRegistration_; 
                }
                else
                {
                      return this.supervisorRegistration_; 
                }
            }
            set
            {
                if (this.supervisorRegistration_ != value)
                {
                    this.supervisorRegistration_ = value;
                }
            }
        }
        #endregion
        
        public string supervisorRegistrationIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-doc-5-4/EndUserExceptionDeclaration.cpp
// EndUserExceptionDeclaration.cpp 
#include "EndUserExceptionDeclaration.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

EndUserExceptionDeclaration::EndUserExceptionDeclaration(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //creditDocumentNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* creditDocumentNode = xmlNode->FirstChildElement("creditDocument");

   if(creditDocumentNode){creditDocumentIsNull_ = false;}
   else{creditDocumentIsNull_ = true;}

   if(creditDocumentNode)
   {
      for(creditDocumentNode; creditDocumentNode; creditDocumentNode = creditDocumentNode->NextSiblingElement("creditDocument")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- creditDocumentNode , address : " << creditDocumentNode << std::endl;
          #endif
          if(creditDocumentNode->Attribute("href") || creditDocumentNode->Attribute("id"))
          {
              if(creditDocumentNode->Attribute("id"))
              {
                  creditDocumentIDRef_ = creditDocumentNode->Attribute("id");
                  creditDocument_.push_back(boost::shared_ptr<CreditDocument>(new CreditDocument(creditDocumentNode)));
                  creditDocument_.back()->setID(creditDocumentIDRef_);
                  IDManager::instance().setID(creditDocumentIDRef_, creditDocument_.back());
              }
              else if(creditDocumentNode->Attribute("href")) { creditDocumentIDRef_ = creditDocumentNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { creditDocument_.push_back(boost::shared_ptr<CreditDocument>(new CreditDocument(creditDocumentNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- creditDocumentNode , address : " << creditDocumentNode << std::endl;
       #endif
   }

   //organizationCharacteristicNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* organizationCharacteristicNode = xmlNode->FirstChildElement("organizationCharacteristic");

   if(organizationCharacteristicNode){organizationCharacteristicIsNull_ = false;}
   else{organizationCharacteristicIsNull_ = true;}

   if(organizationCharacteristicNode)
   {
      for(organizationCharacteristicNode; organizationCharacteristicNode; organizationCharacteristicNode = organizationCharacteristicNode->NextSiblingElement("organizationCharacteristic")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- organizationCharacteristicNode , address : " << organizationCharacteristicNode << std::endl;
          #endif
          if(organizationCharacteristicNode->Attribute("href") || organizationCharacteristicNode->Attribute("id"))
          {
              if(organizationCharacteristicNode->Attribute("id"))
              {
                  organizationCharacteristicIDRef_ = organizationCharacteristicNode->Attribute("id");
                  organizationCharacteristic_.push_back(boost::shared_ptr<OrganizationCharacteristic>(new OrganizationCharacteristic(organizationCharacteristicNode)));
                  organizationCharacteristic_.back()->setID(organizationCharacteristicIDRef_);
                  IDManager::instance().setID(organizationCharacteristicIDRef_, organizationCharacteristic_.back());
              }
              else if(organizationCharacteristicNode->Attribute("href")) { organizationCharacteristicIDRef_ = organizationCharacteristicNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { organizationCharacteristic_.push_back(boost::shared_ptr<OrganizationCharacteristic>(new OrganizationCharacteristic(organizationCharacteristicNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- organizationCharacteristicNode , address : " << organizationCharacteristicNode << std::endl;
       #endif
   }

   //transactionCharacteristicNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* transactionCharacteristicNode = xmlNode->FirstChildElement("transactionCharacteristic");

   if(transactionCharacteristicNode){transactionCharacteristicIsNull_ = false;}
   else{transactionCharacteristicIsNull_ = true;}

   if(transactionCharacteristicNode)
   {
      for(transactionCharacteristicNode; transactionCharacteristicNode; transactionCharacteristicNode = transactionCharacteristicNode->NextSiblingElement("transactionCharacteristic")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- transactionCharacteristicNode , address : " << transactionCharacteristicNode << std::endl;
          #endif
          if(transactionCharacteristicNode->Attribute("href") || transactionCharacteristicNode->Attribute("id"))
          {
              if(transactionCharacteristicNode->Attribute("id"))
              {
                  transactionCharacteristicIDRef_ = transactionCharacteristicNode->Attribute("id");
                  transactionCharacteristic_.push_back(boost::shared_ptr<TransactionCharacteristic>(new TransactionCharacteristic(transactionCharacteristicNode)));
                  transactionCharacteristic_.back()->setID(transactionCharacteristicIDRef_);
                  IDManager::instance().setID(transactionCharacteristicIDRef_, transactionCharacteristic_.back());
              }
              else if(transactionCharacteristicNode->Attribute("href")) { transactionCharacteristicIDRef_ = transactionCharacteristicNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { transactionCharacteristic_.push_back(boost::shared_ptr<TransactionCharacteristic>(new TransactionCharacteristic(transactionCharacteristicNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- transactionCharacteristicNode , address : " << transactionCharacteristicNode << std::endl;
       #endif
   }

   //supervisorRegistrationNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* supervisorRegistrationNode = xmlNode->FirstChildElement("supervisorRegistration");

   if(supervisorRegistrationNode){supervisorRegistrationIsNull_ = false;}
   else{supervisorRegistrationIsNull_ = true;}

   if(supervisorRegistrationNode)
   {
      for(supervisorRegistrationNode; supervisorRegistrationNode; supervisorRegistrationNode = supervisorRegistrationNode->NextSiblingElement("supervisorRegistration")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- supervisorRegistrationNode , address : " << supervisorRegistrationNode << std::endl;
          #endif
          if(supervisorRegistrationNode->Attribute("href") || supervisorRegistrationNode->Attribute("id"))
          {
              if(supervisorRegistrationNode->Attribute("id"))
              {
                  supervisorRegistrationIDRef_ = supervisorRegistrationNode->Attribute("id");
                  supervisorRegistration_.push_back(boost::shared_ptr<SupervisorRegistration>(new SupervisorRegistration(supervisorRegistrationNode)));
                  supervisorRegistration_.back()->setID(supervisorRegistrationIDRef_);
                  IDManager::instance().setID(supervisorRegistrationIDRef_, supervisorRegistration_.back());
              }
              else if(supervisorRegistrationNode->Attribute("href")) { supervisorRegistrationIDRef_ = supervisorRegistrationNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { supervisorRegistration_.push_back(boost::shared_ptr<SupervisorRegistration>(new SupervisorRegistration(supervisorRegistrationNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- supervisorRegistrationNode , address : " << supervisorRegistrationNode << std::endl;
       #endif
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
std::vector<boost::shared_ptr<CreditDocument>> EndUserExceptionDeclaration::getCreditDocument()
{
   if(!this->creditDocumentIsNull_){
        if(creditDocumentIDRef_ != ""){
             return this->creditDocument_;
        }else{
             return this->creditDocument_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<CreditDocument>>();
   }
}
std::vector<boost::shared_ptr<OrganizationCharacteristic>> EndUserExceptionDeclaration::getOrganizationCharacteristic()
{
   if(!this->organizationCharacteristicIsNull_){
        if(organizationCharacteristicIDRef_ != ""){
             return this->organizationCharacteristic_;
        }else{
             return this->organizationCharacteristic_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<OrganizationCharacteristic>>();
   }
}
std::vector<boost::shared_ptr<TransactionCharacteristic>> EndUserExceptionDeclaration::getTransactionCharacteristic()
{
   if(!this->transactionCharacteristicIsNull_){
        if(transactionCharacteristicIDRef_ != ""){
             return this->transactionCharacteristic_;
        }else{
             return this->transactionCharacteristic_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<TransactionCharacteristic>>();
   }
}
std::vector<boost::shared_ptr<SupervisorRegistration>> EndUserExceptionDeclaration::getSupervisorRegistration()
{
   if(!this->supervisorRegistrationIsNull_){
        if(supervisorRegistrationIDRef_ != ""){
             return this->supervisorRegistration_;
        }else{
             return this->supervisorRegistration_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<SupervisorRegistration>>();
   }
}
}

